### PR TITLE
fix lingering refs to 4.x db structure

### PR DESCRIPTION
**Pull request by gilmoskowitz**
and make minor improvements inspired by dogfood customizations

### CODE REVIEW

File: foundation-database/public/tables/metasql/orderActivityByProject-detail.mql [hunk 1]
```diff
@@ -124,15 +126,16 @@ SELECT task_id AS id,
...
+            WHEN task_status = 'O' AND task_due_date BETWEEN current_date AND current_date + (fetchmetricvalue('ProjectDueDateWarning')||' days')::interval THEN 'warning'
          END AS due_qtforegroundrole,
        2 AS xtindentrole
   FROM task
-  LEFT OUTER JOIN te.teprjtask ON (task_id = teprjtask_prjtask_id)
-  LEFT OUTER JOIN custinfo ON (teprjtask_cust_id=cust_id)
-  LEFT OUTER JOIN cntct ON (cust_corrcntct_id=cntct_id)
-  LEFT OUTER JOIN addr ON (cntct_addr_id=addr_id)
+  LEFT OUTER JOIN taskass      ON task_id = taskass_task_id
+  LEFT OUTER JOIN te.teprjtask ON task_id = teprjtask_prjtask_id
```
Comment (ChristopherCotnoir) [inline foundation-database/public/tables/metasql/orderActivityByProject-detail.mql]: This should not be here.

File: foundation-database/public/tables/metasql/orderActivityByProject-detail.mql [hunk 2]
```diff
@@ -115,7 +117,7 @@ SELECT task_id AS id,
        NULL::text AS uom,
        NULL::numeric AS value,
        task_due_date AS due,
-       (SELECT MIN(taskass_assigned_date) FROM taskass WHERE taskass_task_id=task_id) AS assigned,
+       MIN(taskass_assigned_date) AS assigned,
```
Comment (ChristopherCotnoir) [inline foundation-database/public/tables/metasql/orderActivityByProject-detail.mql]: The GROUP BY below is more restrictive, so this won't work.

File: foundation-database/public/tables/metasql/customers-detail.mql [hunk 3]
```diff
@@ -5,15 +5,13 @@
...
-  cust_id, cust_custtype_id, 
+SELECT DISTINCT ON ( cust_number )
+  cust_id, cust_custtype_id,
   cust_active, cust_number, cust_name,
   custtype_code,
   formatBoolYN(cust_active) as f_active,
   formatAddr(bill_addr.addr_id) AS f_bill_addr,
   formatAddr(corr_addr.addr_id) AS f_corr_addr,
-  bill_cntct.cntct_name AS bill_cntct_name,
-  corr_cntct.cntct_name AS corr_cntct_name,
```
Comment (ChristopherCotnoir) [inline foundation-database/public/tables/metasql/customers-detail.mql]: Why are these being removed?

File: foundation-database/public/tables/metasql/projectTasks-detail.mql [hunk 4]
```diff
@@ -8,8 +8,10 @@ SELECT task_id, task_number, task_name,
        COALESCE(task_hours_budget - task_hours_actual, 0.0) as taskhrbal,
        COALESCE(task_exp_budget - task_exp_actual, 0.0) as taskexpbal,
        firstLine(task_descrip) AS task_descrip,
-       totalhrbud, totalhract, totalhrbal,
-       totalexpbud, totalexpact, totalexpbal
+       totalhrbud, totalhract,
+       CASE WHEN task_status = 'C' THEN 0 ELSE totalhrbal END AS totalhrbal,
+       totalexpbud, totalexpact, totalexpbal,
+       0 AS totalhract_xttotalrole
```
Comment (ChristopherCotnoir) [inline foundation-database/public/tables/metasql/projectTasks-detail.mql]: Are we sure these changes are correct?

File: foundation-database/public/tables/metasql/customers-detail.mql (hunk 3 above)
Comment (gilmoskowitz) [reply foundation-database/public/tables/metasql/customers-detail.mql]: These are duplicates of the [billing](https://github.com/xtuple/xtuple/pull/3563/files#diff-c4e9106056c2cd866db791cd76b40607R20) and [correspondence](https://github.com/xtuple/xtuple/pull/3563/files#diff-c4e9106056c2cd866db791cd76b40607R34) contact names below.

File: foundation-database/public/tables/metasql/projectTasks-detail.mql (hunk 4 above)
Comment (gilmoskowitz) [reply foundation-database/public/tables/metasql/projectTasks-detail.mql]: They are correct. The goal seems to be making project totals available for calculations and comparisons with individual task lines. It's unusual but not wrong.

**Comment by gilmoskowitz** [comment]
@ChristopherCotnoir could you please re-review?

**Pull request merged**